tests/maskers/test_tabular.py

```diff
@@ -107,11 +110,103 @@ def test_serialization_partion_masker_numpy():
         temp_serialization_file.seek(0)
 
         # deserialize masker
-        new_partition_masker = shap.maskers.Masker.load(temp_serialization_file)
+        new_partition_masker = shap.maskers.Partition.load(temp_serialization_file)
+
+    mask = np.ones(X.shape[1]).astype(int)
+    mask[0] = 0
+    mask[4] = 0
+
+    # comparing masked values
+    assert np.array_equal(original_partition_masker(mask, X[0])[0], new_partition_masker(mask, X[0])[0])
+
+def test_serialization_impute_masker_dataframe():
+    """ Test the serialization of a Partition masker based on a DataFrame.
+    """
+
+    X, _ = shap.datasets.california(n_points=500)
+
+    # initialize partition masker
+    original_partition_masker = shap.maskers.Impute(X)
+
+    with tempfile.TemporaryFile() as temp_serialization_file:
+
+        # serialize partition masker
+        original_partition_masker.save(temp_serialization_file)
+
+        temp_serialization_file.seek(0)
+
+        # deserialize masker
+        new_partition_masker = shap.maskers.Impute.load(temp_serialization_file)
+
+    mask = np.ones(X.shape[1]).astype(int)
+    mask[0] = 0
+    mask[4] = 0
+
+    # comparing masked values
+    assert np.array_equal(original_partition_masker(mask, X[:1].values[0])[1], new_partition_masker(mask, X[:1].values[0])[1])
+
+def test_serialization_impute_masker_numpy():
+    """ Test the serialization of a Partition masker based on a numpy array.
+    """
+
+    X, _ = shap.datasets.california(n_points=500)
+    X = X.values
+
+    # initialize partition masker
+    original_partition_masker = shap.maskers.Impute(X)
+
+    with tempfile.TemporaryFile() as temp_serialization_file:
+
+        # serialize partition masker
+        original_partition_masker.save(temp_serialization_file)
+
+        temp_serialization_file.seek(0)
+
+        # deserialize masker
+        new_partition_masker = shap.maskers.Impute.load(temp_serialization_file)
 
     mask = np.ones(X.shape[1]).astype(int)
     mask[0] = 0
     mask[4] = 0
 
     # comparing masked values
     assert np.array_equal(original_partition_masker(mask, X[0])[0], new_partition_masker(mask, X[0])[0])
+
+def test_imputation():
```

Comment:
I think we should get another test where we have values to impute (so explicitly set some values to `np.nan`).

Reply:
also can we do 

```python
@pytest.mark.parametrize("method", ["linear", "mean", "median", "most_frequent", "knn"])
def test_imputation(method):
    ...
```
and then get rid of the loop within the function? That helps us locate what method throws an error